apps/web/app/(app)/mail/page.tsx

```diff
@@ -20,65 +20,80 @@ export default function Mail({
     ? { type: searchParams.type }
     : {};
 
-  const { data, isLoading, error, mutate } = useSWR<ThreadsResponse>(
-    `/api/google/threads?${new URLSearchParams(query as any).toString()}`,
-    {
+  const getKey = (
+    pageIndex: number,
+    previousPageData: ThreadsResponse | null,
+  ) => {
+    if (previousPageData && !previousPageData.nextPageToken) return null;
+    // For the first page, use the base query
+    if (pageIndex === 0)
+      return `/api/google/threads?${new URLSearchParams(query as Record<string, string>).toString()}`;
+    // For subsequent pages, append the nextPageToken
+    return `/api/google/threads?nextPageToken=${previousPageData?.nextPageToken}&${new URLSearchParams(query as any).toString()}`;
+  };
```

Comment:
cleaner way to do this is without the if/else around the whole row as most of it is duplicated.

something like this:

```ts
const getKey = (pageIndex: number, previousPageData: ThreadsResponse | null) => {
  if (previousPageData && !previousPageData.nextPageToken) return null;

  const queryParams = new URLSearchParams(query as Record<string, string>);
  
  if (previousPageData?.nextPageToken) {
    queryParams.set('nextPageToken', previousPageData.nextPageToken);
  }

  return `/api/google/threads?${queryParams.toString()}`;
};
```

Reply:
Done!